tket-py/examples/utils/__init__.py

```diff
@@ -27,14 +27,15 @@ def _repr_tk2circ(
 
 # TODO: Should this be part of the guppy API? Or tket?
 # Takes a RawFunctionDef and converts it to a Tk2Circuit
+# BUG: this defaults num_operations to 0 which breaks Rules
 def guppy_to_circuit(func_def: Any) -> Tk2Circuit:
     """Convert a Guppy function definition to a `Tk2Circuit`."""
-    module = func_def.id.module
-    assert module is not None, "Function definition must belong to a module"
 
-    pkg = module.compile()
+    pkg = func_def.compile_function()
 
-    json = pkg.package.to_str(EnvelopeConfig.TEXT)
-    circ = Tk2Circuit.from_str(json, func_def.name)
+    f_name = pkg.modules[0].entrypoint_op().f_name
+
+    json = pkg.to_str(EnvelopeConfig.TEXT)
+    circ = Tk2Circuit.from_str(json, f_name)
```

Comment:
I think this similarly can be 


```suggestion
    byt = pkg.to_bytes()
    circ = Tk2Circuit.from_bytes(byt, f_name)
```

Reply:
yeah that also works, committed now